Add baggage span processor

martin308 opened the pull request:
Adds a baggage span processor as per the other contrib repositories.

This processor takes items from the baggage and adds the items as attributes on spans.

Eg. https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5404

martin308 commented:
The macOS tests look like they might be flaky but the linux test result is for my new tests will poke around and try to find the cause

nachoBonafonte reviewed (changes requested):
Could you change it to a more expressible naming, it looks very generic like this
It also would need proper documentation in the source code for the future  users, for the rest it LGTM

martin308 commented:
From the sync, naming wise we will go `BaggagePropagationProcessor`

martin308 commented:
Updated with the following changes:
- new name for the processor
- moved to an individual library/target for the processor

I left the contrib tests as a test target but we could break those up into individual test targets too if we think it is worth it?

nachoBonafonte reviewed (approved):
Looks good to e, I don't mind having the contribution tests under the same test target or separated
